client/my-sites/controller.js

```diff
@@ -310,15 +310,20 @@ export function siteSelection( context, next ) {
 	const hasOneSite = currentUser.visible_site_count === 1;
 	const allSitesPath = sectionify( context.path, siteFragment );
 
+	// See https://github.com/Automattic/wp-calypso/issues/23517
+	// Avoid recording non-normalized paths in page view stats (e.g. /post/1)
+	const pathParts = basePath.split( '/' );
+	const rootPath = pathParts.length > 2 ? `/${ pathParts[ 1 ] }` : basePath;
+
 	if ( currentUser && currentUser.site_count === 0 ) {
 		renderEmptySites( context );
-		return analytics.pageView.record( basePath, sitesPageTitleForAnalytics + ' > No Sites' );
+		return analytics.pageView.record( rootPath, sitesPageTitleForAnalytics + ' > No Sites' );
```

Comment:
As an alternative, could we just hard-code `/sites` here because the original section wouldn't be reached anyway?

Reply:
@yanirs what's your preference here?

Another option might be `/post/:site/:post_id` and note that the user has no sites?

Reply:
What does the user actually see? If I understand correctly, they don't get to `/post/:site/:post_id` since they have no sites, so something like `/sites` makes sense.

Reply:
> If I understand correctly, they don't get to /post/:site/:post_id since they have no sites.

That's correct, they end up on a blank page with "no visible sites" message.

Reply:
Updated as suggested in b4e89a8a591a010d83f8d874f247a31cb864342c.